### DISCUSSION TITLE
Netherlands map behaves erratically

### DISCUSSION THREAD

**Discussion by joostbijl** [devemux86/cruiser#210] [Q&A] [answered]
When panning or zooming the map, around zoom level 9 the map starts to behave erratically like in the video. After some time (10-30s) it behaves normally until you pan again.

https://youtube.com/shorts/gFdMc7uzfmE?feature=share

This is the latest cruiser (1 jun) on a bv5100

Nay thoughts on what may cause. This?

**Comment by devemux86** [comment]
@joostbijl thanks for the report!

Indeed this is strange, I have never seen it before.

What happens if you restart the phone? Can you reproduce it each time?

Can you share the app settings .pref file to see what settings are enabled?

In case you have changed Cruiser settings, can you reset them and try again?

**Comment by devemux86** [comment]
@joostbijl 

Is the phone connected with bluetooth with an external device?

Is a device selected in "External input device" in Cruiser settings?
Maybe it is interfering with the map zoom / move?

What happens if you select "None" in "External input device" in Cruiser settings?
Or if you turn off bluetooth?

**Comment by joostbijl** [comment]
I sent you my prefs. File by mail. I did have a remote controller attached, but the same behavior happens if I turn that off, disable Bluetooth or set the remote keyboard to 'none'.

I suspect it has to do with the speed of the device, it is pretty slow

**Reply by devemux86** [reply]
@joostbijl thanks for the .pref file!

I examined it, all settings seem normal and should not affect the map in such way.

What happens if you restart the phone? Also can you reproduce it each time?

Does it always happen at same zoom? What happens if you use an online map?

**Reply by joostbijl** [reply]
> What happens if you restart the phone? Also can you reproduce it each time?

Yes, makes no difference

> Does it always happen at same zoom? 

It seems to be less of an issue at low and high zoom levels. At zoom level 8-10 it seems worst. 

> What happens if you use an online map?

Then it works well.

I also deleted the maps, and downloaded them again, but that did not solve the issue. 

I'll now remove the app and reinstall and redownload the maps.

**Reply by joostbijl** [reply]
Unfortunately, a reinstall has the same issue.

**Reply by devemux86** [reply]
@joostbijl 

> What happens if you use an online map?
> Then it works well.

This is interesting.

Then please load only 1 offline map and mention what theme you use.

For example activate 1 Mapsforge map with the default Biker theme.
You can also try to disable Vegetation in map layers.

**Reply by devemux86** [reply]
@joostbijl 

Also please try without other apps running that can affect the screen overlay,
like apps for recording, speed limit, weather, etc.

i.e. restart the phone and run only Cruiser to be sure.

Such apps can often affect the screen rendering on Android.

**Reply by joostbijl** [reply]
hey @devemux86 , 

I used only the offline map (mapsforge) with the Biker theme. I also tried other themes: no difference.

I did have 'automate'  installed but inactive. I removed it, but made no difference.

I then tried a few different maps: Luxemburg (because it's small), and ' England' (because it's about the same size as NL in MB). Both work smooth. 

I also deactived/deleted poi and graphhopper data, but made no difference.

So it seems that it's a problem with the Netherlands mapsforge map in offline mode. 


If you just do that, can you reproduce it?

**Reply by devemux86** [reply]
@joostbijl 

> I then tried a few different maps: Luxemburg (because it's small), and ' England' (because it's about the same size as NL in MB). Both work smooth.

> So it seems that it's a problem with the Netherlands mapsforge map in offline mode.

Netherlands is one of the heavier maps to render due to its complex nature and water polygons.
But there is an extra improvement that I can make in the performance of the offline maps. :slightly_smiling_face: 

Can you please turn off "Vegetation" in map layers and test again?

**Reply by joostbijl** [reply]
> Netherlands is one of the heavier maps to render due to its complex nature and water polygons. But there is an extra improvement that I can make in the performance of the offline maps. 🙂

But it doesn't happen in london, that is also pretty dense.
 
> Can you please turn off "Vegetation" in map layers and test again?

Sorry, forgot to mention that. I did turn it off, but made no difference

**Reply by devemux86** [reply]
> But it doesn't happen in london, that is also pretty dense.

It is not the same type of content. :slightly_smiling_face: 

It is about nature / water polygons that are very complex in Netherlands.
Some graphics cards (even on desktop) may have issues with all these.

I plan to add an extra performance improvement to help in such cases.

**Reply by joostbijl** [reply]
> > But it doesn't happen in london, that is also pretty dense.
> 
> It is not the same type of content. 🙂
> 
> It is about nature / water polygons that are very complex in Netherlands. Some graphics cards (even on desktop) may have issues with all these.
> 
> I plan to add an extra performance improvement to help in such cases.

So if i were to do the same steps on a faster phone, it would be OK? I wouldn't mind if the waterways are not displayed 100% correct, as long as the roads are ;). 

Is there anything else i can do to support the hypothesis? I can load another water-rich area of the world?

**Reply by devemux86** [reply]
@joostbijl 

That's why we have the Vegetation as layer, because in navigation we are mainly interested for the roads.

I can also make the water polygons as layer and you could try in your device if it improves Netherlands?

**Reply by joostbijl** [reply]
> @joostbijl
> 
> That's why we have the Vegetation as layer, because in navigation we are mainly interested for the roads.
> 
> I can also make the water polygons as layer and you could try in your device if it improves Netherlands?


Sure, for me that's easy to try (but you have to do the hard work). If you need me to install a development build let me know!

**Reply by devemux86** [reply]
@joostbijl 

Please try the attached Biker map theme: [biker.xml.zip](https://github.com/devemux86/cruiser/files/11731744/biker.xml.zip)
- Select "Settings | Application | Advanced | Menu | Advanced"
- Use menu "Map | Open map theme" to open the theme xml
- Turn off the "Water regions" with menu "Map | Map layers"

**Reply by joostbijl** [reply]
> * Turn off the "Water regions" with menu "Map | Map layers"

Yes, this is reproducible. If I turn it on, it behaves erratically. If I turn it off the map doesn't misbehave but is still a bit slowish (slower than online maps). The bodies of water that don't show are now a whole different view of the region haha. 

The polygon fix you mentioned, is that a rendering fix, or a map theme/style fix?

**Reply by devemux86** [reply]
> The polygon fix you mentioned, is that a rendering fix, or a map theme/style fix?

A rendering performance improvement. But turn off water regions is still the fastest improvement.

If the map seems slow without vegetation and water areas, then maybe it is a problem with the device?
Probably the graphics card. OpenGL is fast and should not be any problems unless overload the map.

Info: offline maps are rendered on the device using the phone cpu and graphics card.

**Reply by devemux86** [reply]
@joostbijl 

You need to turn off both vegetation + water to test the speed on your device.

However even on slow devices, OpenGL should work fast without problems.

**Reply by devemux86** [reply]
@joostbijl 

Also make sure you have selected the "Settings | Map | Advanced | Rendering (OpenGL) | **Performance**".

**Comment by devemux86** [comment]
The improvement for the water polygons will be available in the next Cruiser update.

**Comment by joostbijl** [comment]
> Also make sure you have selected the "Settings | Map | Advanced | Rendering (OpenGL) | **Performance**".

I do have that selected, but it is still a tad slow. I guess that's just the limit of the hardware. It is not a powerhouse: https://www.gsmarena.com/blackview_bv5100-11567.php

**Comment by devemux86** [comment, marked as answer]
> The improvement for the water polygons will be available in the next Cruiser update.

Available in [Cruiser 3.0.24](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by joostbijl** [reply]
This version works good on my phone! Even with water rendering ;)
